src/CMakeLists.txt

```diff
@@ -84,12 +81,20 @@ if(CMAKE_INTERPROCEDURAL_OPTIMIZATION)
     # Policy that enables INTERPROCEDURAL_OPTIMIZATION
     set(CMAKE_POLICY_DEFAULT_CMP0069 NEW)
     cmake_policy(SET CMP0069 NEW)
-    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
     if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
       # Clang has some issues with thinLTO (segfaults on final linking)
       # This enables full LTO through CMake
       # FullLTO is slow, but may yield performance improvements.
-      add_compile_options(-flto=full)
+      #set(CMAKE_{lang}_COMPILE_OPTIONS_IPO ${CMAKE_CXX_COMPILE_OPTIONS_IPO} -flto)
+      add_compile_options(-flto) # alternate way of setting LTO.
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+      # If we are on GCC, override CMake do parallel LTO.
+      include(ProcessorCount)
+      ProcessorCount(N_LTO_JOBS)
+      # Flag must only be passed to linker to prevent clang-tidy errors
+      string(REPLACE "<LINK_FLAGS>" "<LINK_FLAGS> -flto=${N_LTO_JOBS}"
+                     CMAKE_CXX_LINK_EXECUTABLE "${CMAKE_CXX_LINK_EXECUTABLE}"
```

Comment:
I don't recommend doing this, it's very brittle and consequences of changing this mid-build are unknown. It might work with some compilers / on some platforms and fail spectacularly on other. It should only get changed in the toolchain files, not in the build scripts themselves.

(What's the use case of having both Clang Tidy and LTO enabled? Is that common? I'd say the cases where Clang Tidy is enabled (locally and on the CI) don't need LTO as well, and where LTO is used you don't want to have your build slowed down even further by Clang Tidy.)

Reply:
We run Clang-Tidy separately as part with run-clang-tidy not as part of the CMake, you just need to generate another compile.json without the flto=N on the compiler args (since it has no affect there anyhow). @mosra We aren't changing it mid build though? This setting it for downstream builds?